src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java

```diff
@@ -64,136 +55,65 @@ public CreateCaseCallbackService(
-        } catch (Exception exception) {
-            log.error(
-                "Failed to create exception for service {} and exception record {}",
-                configItem.getService(),
-                caseIdAsString,
-                exception
-            );
-
-            return Validation.invalid(Array.of("Internal error. " + exception.getMessage()));
+        Validation<String, Void> event = isCreateNewCaseEvent(request.getEventId());
+        if (event.isInvalid()) {
+            return new ProcessResult(emptyList(), singletonList(event.getError()));
+        } else {
+            return hasServiceNameInCaseTypeId(request.getCaseDetails())
+                .map(serviceName -> {
+                    try {
+                        ServiceConfigItem serviceCfg = serviceConfigProvider.getConfig(serviceName);
+                        if (serviceCfg == null || serviceCfg.getTransformationUrl() == null) {
+                            return new ProcessResult(emptyList(), singletonList("Transformation URL is not configured"));
+                        } else {
+                            return validator
+                                .getValidation(request.getCaseDetails())
+                                .map(exceptionRecord -> {
+                                    SuccessfulTransformationResponse transformationResp =
+                                        transformationClient.transformExceptionRecord(
+                                            serviceCfg.getTransformationUrl(),
+                                            exceptionRecord,
+                                            s2sTokenGenerator.generate()
+                                        );
+                                    long newCaseId = createNewCaseInCcd(
+                                        idamToken,
+                                        userId,
+                                        exceptionRecord.poBoxJurisdiction,
+                                        transformationResp.caseCreationDetails,
+                                        request.getCaseDetails().getId().toString()
+                                    );
+                                    return new ProcessResult(ImmutableMap.of(CASE_REFERENCE, Long.toString(newCaseId)));
+                                })
+                                .getOrElseGet(errors -> new ProcessResult(emptyList(), errors.asJava()));
+                        }
+                    } catch (ServiceNotConfiguredException exc) {
+                        return new ProcessResult(emptyList(), singletonList(exc.getMessage()));
+                    } catch (InvalidCaseDataException exc) {
+                        return new ProcessResult(exc.getResponse().warnings, exc.getResponse().errors);
+                    } catch (Exception exc) {
+                        log.error("Error handling event", exc);
+                        return new ProcessResult(emptyList(), singletonList("Internal error"));
```

Comment:
500, too

Reply:
Let's keep those notes for later, but this PR is strictly about refactoring, keeping existing tests passing.